src/awkward/_v2/_connect/rdataframe/to_rdataframe.py

```diff
@@ -6,8 +6,20 @@
 import awkward._v2._connect.cling  # noqa: E402
 
 import ROOT
+import threading
 
 compiler = ROOT.gInterpreter.Declare
+compiler_lock = threading.Lock()
+compiler(
+    """
+#include <Python.h>
+"""
+)
+
+
+def compile(source_code):
+    with compiler_lock:
+        return compiler(source_code)
```

Comment:
Might as well be consistent and use the lock for the `#include <Python.h>` as well as everything else.

Also, no reason to introduce the name "`compiler`" distinct from "`compile`". It's not used anywhere else.

```suggestion
def compile(source_code):
    with compiler_lock:
        return ROOT.gInterpreter.Declare(source_code)


compiler_lock = threading.Lock()
compiler("#include <Python.h>\n")
```

Reply:
done